offchain-modules/packages/x/src/ckb/tx-helper/generator.ts

```diff
@@ -802,6 +802,223 @@ export class CkbTxGenerator extends CkbTxHelper {
 
     return txSkeleton;
   }
+
+  async unlockSUDT(
+    sudtArgs: string,
+    records: { burnId: string; recipient: string; amount: string }[],
+  ): Promise<TransactionSkeletonType> {
+    const collectorAddress = getFromAddr();
+
+    let txSkeleton = TransactionSkeleton({
+      cellProvider: this.indexer,
+    });
+
+    // search admin cell
+    const adminCell = await this.fetchOmniLockAdminCell();
+    const adminCellOutpoint = adminCell?.out_point;
+    if (!adminCellOutpoint) throw new Error('omni lock admin cell outpoint could not be null');
+    const adminCellCellDep: CellDep = {
+      out_point: adminCellOutpoint,
+      dep_type: 'code',
+    };
+
+    // add cellDeps
+    txSkeleton = txSkeleton.update('cellDeps', (cellDeps) => {
+      const found = nonNullable(this.lumosConfig.SCRIPTS.SECP256K1_BLAKE160);
+      return cellDeps.push({
+        dep_type: found.DEP_TYPE,
+        out_point: { tx_hash: found.TX_HASH, index: found.INDEX },
+      });
+    });
+    txSkeleton = txSkeleton.update('cellDeps', (cellDeps) => {
+      return cellDeps.push(adminCellCellDep);
+    });
+    txSkeleton = txSkeleton.update('cellDeps', (cellDeps) => {
+      const omniLockCellDep = nonNullable(ForceBridgeCore.config.ckb.deps.omniLock?.cellDep);
+      return cellDeps.push({
+        out_point: {
+          tx_hash: omniLockCellDep.outPoint.txHash,
+          index: omniLockCellDep.outPoint.index,
+        },
+        dep_type: omniLockCellDep.depType,
+      });
+    });
+
+    const sudtScript = {
+      code_hash: ForceBridgeCore.config.ckb.deps.sudtType.script.codeHash,
+      hash_type: ForceBridgeCore.config.ckb.deps.sudtType.script.hashType,
+      args: sudtArgs,
+    };
+
+    let totalSudtOutputCapacity = BigInt(0);
+    let totalSudtOutputAmount = BigInt(0);
+    // add recipient outputs
+    records.forEach((record) => {
+      const amount = BigInt(record.amount);
+      const recipientCell: Cell = {
+        cell_output: {
+          capacity: '0x0',
+          lock: parseAddress(record.recipient),
+          type: sudtScript,
+        },
+        data: utils.toBigUInt128LE(amount),
+      };
+      const recipientCellMinimalCapacity = minimalCellCapacity(recipientCell);
+      totalSudtOutputCapacity += recipientCellMinimalCapacity;
+      totalSudtOutputAmount += amount;
+      recipientCell.cell_output.capacity = `0x${recipientCellMinimalCapacity.toString(16)}`;
```

Comment:
recipientCell capacity is a fixed number from config, ref:

https://github.com/nervosnetwork/force-bridge/blob/85725f19cae4f5c36eacf8e351e85568ac56c6fa/offchain-modules/packages/x/src/ckb/tx-helper/generator.ts#L247

Reply:
Useful knowledge acquired